tokio-trace-fmt/src/filter/reload.rs

```diff
@@ -0,0 +1,149 @@
+use parking_lot::RwLock;
+use std::{
+    error, fmt,
+    sync::{Arc, Weak},
+};
+use tokio_trace_core::{subscriber::Interest, Metadata};
+use {filter::Filter, span::Context};
+
+#[derive(Debug)]
+pub struct ReloadFilter<F> {
+    inner: Arc<RwLock<F>>,
+}
+
+#[derive(Debug, Clone)]
+pub struct Handle<F> {
+    inner: Weak<RwLock<F>>,
+}
+
+#[derive(Debug)]
+pub struct Error {
+    kind: ErrorKind,
+}
+
+#[derive(Debug)]
+enum ErrorKind {
+    SubscriberGone,
+}
+
+// ===== impl ReloadFilter =====
+
+impl<F, N> Filter<N> for ReloadFilter<F>
+where
+    F: Filter<N>,
+{
+    fn callsite_enabled(&self, _: &Metadata, _: &Context<N>) -> Interest {
+        // TODO(eliza): When tokio-rs/tokio#1039 lands, we can allow our
+        // interest to be cached. For now, we must always return `sometimes`.
+        Interest::sometimes()
+    }
+
+    fn enabled(&self, metadata: &Metadata, ctx: &Context<N>) -> bool {
+        self.inner.read().enabled(metadata, ctx)
+    }
+}
+
+impl<F: 'static> ReloadFilter<F> {
```

Comment:
im not totally following why this needs `'static`?

Reply:
@LucioFranco it's because of the `'static` bound on the filter type on the impl of `Subscriber` for `FmtSubscriber`:
https://github.com/tokio-rs/tokio-trace-nursery/blob/462a08fb6ca64db3944689ecafdae0107c6982c7/tokio-trace-fmt/src/lib.rs#L83
this is because implementors of `Subscriber` have to be `'static`. So a non-`'static` filter wouldn't actually be _usable_.

In retrospect, the `'static` bound should _probably_ just be on the `Filter` trait...

Reply:
Right, that makes sense, it also makes sense to move it lower into the Filter trait.